shared/schemas/projects/create-project.schema.ts

```diff
@@ -8,19 +8,25 @@ import { z } from "zod";
 
 export const CreateProjectSchema = z.object({
   projectName: z
-    .string()
-    .min(3, {
-      message: "Name must contain at least 3 characters.",
-    })
-    .max(255, { message: "Name must be less than 255 characters" }),
+      .string()
+      .min(3, { message: "Name must contain at least 3 characters." })
+      .max(255, { message: "Name must be less than 255 characters" }),
   countryCode: z.string().min(3).max(3),
   ecosystem: z.nativeEnum(ECOSYSTEM),
   activity: z.nativeEnum(ACTIVITY),
   projectSizeHa: z.number().nonnegative(),
-  priceType: z.nativeEnum(PROJECT_PRICE_TYPE),
+  priceType: z.nativeEnum(PROJECT_PRICE_TYPE).optional(),
   initialCarbonPriceAssumption: z.number().nonnegative(),
   restorationActivity: z
-    .nativeEnum(RESTORATION_ACTIVITY_SUBTYPE)
-    .optional()
-    .nullable(),
+      .nativeEnum(RESTORATION_ACTIVITY_SUBTYPE)
+      .optional()
+      .nullable(),
+}).superRefine((data, ctx) => {
+  if (data.activity === ACTIVITY.RESTORATION && !data.restorationActivity) {
```

Comment:
This MUST be required for Restoration type Projects, otherwise the output does not make sense. Not sure how this will play in the backoffice